### ISSUE TITLE
Unnecessary GPU (D3D) load from running FFmetrics (while main window is not minimized)

### ISSUE THREAD

**Issue by jeryll7** [fifonik/FFMetrics#109]
I recently noticed that this great and useful program (thank you for it!) is unfortunately causing unnecessary GPU (D3D) load (around 20-30%) on my Nvidia RTX 3070 graphic card. I suspect it has something to do with small preview window next to "Browse" button. Can you check this with HWINFO64 or GPU-Z utility on your system and confirm?

I am mostly concerned about energy consumption of the PC in the long term (since my PC is often running 24/7). Monitoring programs shows around 2 Watt per second increase in graphic card consumption (that's 7.2 kW per hour :( (if I forget to minimize it and leave it running in the background)).

Solution for me (for now - until hopefully fixed ASAP) is relatively simple, if you minimize FFmetrics into taskbar energy consumptions drops back.

But still would be nice this have fixed.

**Comment by fifonik** [comment]
Thanks for your report.
I will try to debug what is causing the high GPU usage.

**Comment by fifonik** [comment]
The issue should be resolved in 1.4.7 beta 1 that I've just released.
Could you check it please?
Thanks.

**Comment by jeryll7** [comment]
Thank you for the quick fix, **I can confirm that unnecessary GPU load is fixed in 1.4.7 beta 1**. But now I get occasional exception thrown at the start of FFmetrics 1.4.7 beta 1:

_CiceroUIWndFrame: FFMetrics.exe - Application Error : The exception unknown software exception (0xe0434352) occurred in the application at location 0x00007FFB865BCD29._

and can't properly compute VMAF score with vmaf_0.6.1- (vmaf_4k_0.6.1 works OK). Can't restart my PC for the next 6 hours or so (running transcode), so for now, I'll have to wait if after restart issue will be gone.

Or maybe you can release quick GPU fix for the v 1.3.1, until betas are polished enough?

**Comment by fifonik** [comment]
> now I get occasional exception thrown at the start of FFmetrics 1.4.7 beta 1

So there is no such startup exception in 1.4.6b?


>  can't properly compute VMAF score with vmaf_0.6.1

Meaning? Error? Wrong score?

**Comment by jeryll7** [comment]
- ok, so I found a GPU bug on v 1.3.1 stable, for whatever reason I missed all the beta releases...
- just for the sake of testing I downloaded all the betas and the first that starts throwing (the same) exceptions is 1.4.2b
- as for the latest incorrect calculations of vmaf score - the problem lies in the added -r switch into FFmpeg command line :
`2023-02-18 17:25:44	ffmpeg -hide_banner -nostdin -r 23.81 -i reencoded.mkv -r 23.98 -i original.mkv -frames:v 480 -lavfi [0:v]trim=start=60,settb=AVTB,setpts=PTS-STARTPTS[main];[1:v]trim=start=60,settb=AVTB,setpts=PTS-STARTPTS[ref];[main][ref]libvmaf=eof_action=endall:n_threads=23:log_fmt=json:log_path='Z\:/TMP/FFMetrics/825e1c2d-64cb-455e-be3c-75a57c515c8b.log':pool=Mean:model=version=vmaf_v0.6.1 -f null -`
-- you are probably extracting frame rate info via FFmpeg focusing on video stream, and I'm not sure its (FFmpeg frame rate detection from video stream) working correctly (at least for Matroska (.mkv) format)
-- I've done only limited number of tests with 1.4.7b1, but a lot of my reencoded MKVs are VFRs with 23.976 fps and all of them are (also) detected by FFmpeg family based players, encoders, decoders as 23.81 fps (I noticed this while using FFmpeg, FFprobe, and MPV player)
-- I'm no coder, nor expert on Matroska - but here is interesting debate about frame rate detection in ffmpeg [https://trac.ffmpeg.org/ticket/259](url) - and If I understand that thread correctly, nothing were fixed (that was 8 years ago!) 
-- in my simple view of a home user, you can compute correct frame rate of media file through "frames/duration" formula or switch to Mediainfo (cli or dll - whatever suits you (so far Mediainfo always show frame rate of my medias correctly)) to get the correct fps of the analyzed media
-- or make use of "-r" switch optional

**Comment by fifonik** [comment]
It will be hard to follow a number of different not related to the title issues here.
Could you please create separate issues for exception and for the wrong vmaf value please?
Thanks.

**Comment by jeryll7** [comment]
Of course
